src/ChannelServer/World/SpawnZone.cs

```diff
@@ -0,0 +1,219 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Melia.Shared.Data.Database;
+using Melia.Shared.Util;
+using Melia.Shared.World;
+
+namespace Melia.Channel.World
+{
+
+
+	public struct SpawnData
+	{
+		public string spawnName;
+		public string mapName;
+		public string monsterName;
+		public int count;
+		public int countVariation;
+		public bool isFixedLocation;
+		public Position Position { get; set; }
+		public Direction Direction { get; set; }
+		public int respawnTime;
+		public int xMin, xMax, zMin, zMax;
+	}
+
+	public class SpawnZone
+	{
+
+		public int Id;
+		public SpawnData Data { get; set; }
+		public MonsterData MonsterData { get; set; }
+		private Map _map;
+		private Dictionary<int, Monster> _spawnEntities;
+		private int _calculatedTotalEntities;
+		private int _countEntities;
+		private int _countDeaths;
+		private bool _isInitialized;
+		private bool _respawnEnabled;
+		private Random rndGenerator;
+
+		public SpawnZone(SpawnData spawnData)
+		{
+			this.Data = spawnData;
+			this.MonsterData = ChannelServer.Instance.Data.MonsterDb.Find(this.Data.monsterName);
+
+			if (this.MonsterData == null)
```

Comment:
It would be better to (also) verify this when the spawn zone is created from the script, to let the scripter know immediately that the monster doesn't exist. Otherwise they potentially have to go search for which script even created this spawner.